spec/controllers/presentations_controller_spec.rb

```diff
@@ -49,45 +143,72 @@
       end
     end
   end
-  context 'the current person is a non-admin' do
-    let(:non_admin) { FactoryGirl.build(:person) }
 
-    before { mock_pundit_user_as(non_admin) }
+  describe 'PATCH #update' do
+    subject do
+      patch :update, params: { id: presentation.id, presentation: attr }
+    end
+    let(:presentation) { FactoryGirl.create(:presentation) }
 
-    describe 'GET #show' do
-      subject { get :show, params: { id: presentation.id } }
-      let!(:presentation) { FactoryGirl.create(:presentation) }
+    context 'the current person is a non-admin' do
+      before { mock_pundit_user_as(non_admin) }
+      let(:attr) { { topic: 'New Topic' } }
 
-      it { is_expected.to render_template :show }
-      it { is_expected.to be_successful }
+      it 'throws a not authorized error' do
+        expect { subject }.to raise_error(Pundit::NotAuthorizedError)
+      end
     end
-  end
 
-  context 'the current person is not signed in' do
-    let(:nil_person) { nil }
+    context 'the current person is an admin' do
+      before { mock_pundit_user_as(admin_person) }
 
-    before { mock_pundit_user_as(nil_person) }
+      context 'changing topic' do
+        let(:attr) { { topic: 'New Topic' } }
 
-    describe 'POST #create' do
-      subject { post :create, params: { presentation: params } }
-      let(:event) { FactoryGirl.create(:event) }
-      let(:params) do
-        FactoryGirl.attributes_for(
-          :presentation,
-          event_id: event.id
-        )
+        it { is_expected.to redirect_to presentation }
+
+        it 'should update presentation' do
+          expect { subject }
+            .to change { presentation.reload.topic }
+            .from('Being Awesome')
+            .to('New Topic')
+        end
+      end
+
+      context 'invalid attributes' do
+        let(:attr) { { topic: nil } }
+
+        it { is_expected.to render_template :edit }
 
-        it 'throws a not authorized error' do
-          expect { subject }.to raise_error(Pundit::NotAuthorizedError)
+        it 'shouldn\'t update presentation' do
+          expect { subject }
+            .to_not change { presentation.reload.topic }
+            .from('Being Awesome')
         end
       end
     end
+  end
 
-    describe 'GET #index' do
-      subject { get :index }
+  describe 'DELETE #destroy' do
+    subject { delete :destroy, params: { id: presentation.id } }
+    let!(:presentation) { FactoryGirl.create(:presentation) }
```

Comment:
As discussed, the way this test is implemented it doesn't work without the bang because when presentation gets called in the subject block it immediately destroys the presentation, and so the test to check if the presentation count has changed will fail.